### ISSUE TITLE
v3.12 baffling conflict issue with NUnit.Engine, IConfiguration on NET5

### ISSUE THREAD

**Issue by replaysMike** [nunit/nunit-console#1027]
I ran into the weirdest reference problem which was eventually identified by using NUnit.Engine 3.12 along with SignalR and a another configuration package on net5. Hang with me here, but it seems when referencing NUnit.Engine 3.12 one of the packages dies with a very specific message: `System.MissingMethodException : Method not found: 'Microsoft.Extensions.Primitives.IChangeToken Microsoft.Extensions.Configuration.IConfiguration.GetReloadToken()'.` When referencing NUnit.Engine 3.11 the error is no longer present. The source package `Any-Config` has an implementation and works fine except under this very specific scenario, and it's not clear how NUnit.Engine is possibly even having this type of effect on it's implementation.

Here is a full repro example:

UnitTest1.cs:
```
public class Tests
{
    [Test]
    public void Test1()
    {
        var value = AnyConfig.Config.Get<bool>("TestBool");
        Assert.AreEqual(true, value);
    }
}
```

appsettings.json:
```:json
{
  "TestBool": true
}
```

TestProject.csproj:
```
<Project Sdk="Microsoft.NET.Sdk">

	<PropertyGroup>
		<TargetFramework>net5.0</TargetFramework>
		<IsPackable>false</IsPackable>
	</PropertyGroup>

	<ItemGroup>
		<FrameworkReference Include="Microsoft.AspNetCore.App" />
	</ItemGroup>

	<ItemGroup>
		<!-- Changing to NUnit.Engine 3.11.0 causes the test to pass -->
		<PackageReference Include="NUnit.Engine" Version="3.12.0" />

		<PackageReference Include="Microsoft.AspNetCore.SignalR.Client" Version="5.0.11" />
		<PackageReference Include="Any-Config" Version="1.0.155" />
		<PackageReference Include="NUnit" Version="3.12.0" />
		<PackageReference Include="NUnit3TestAdapter" Version="3.16.1" />
		<PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.5.0" />
	</ItemGroup>

	<ItemGroup>
		<None Update="appsettings.json">
			<CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
		</None>
	</ItemGroup>

</Project>
```

I'd love to hear your thoughts on what could possibly be going wrong here. Was removing Mono.Cecil without risk? It seems like something is overriding type loading in one of the assemblies or something, I really can't even explain how this is possible it's breaking the way it is.

**Comment by CharliePoole** [comment]
@replaysMike Is this still an active issue with 3.13? If so, what is the command-line, which causes it?

**Comment by replaysMike** [comment]
I'll check today. We had downgraded for the moment to move past it, but I will investigate again

**Comment by CharliePoole** [comment]
Also... we just released 3.14 as well.

**Comment by CharliePoole** [comment]
@replaysMike Well, we're now at 3.15, which was released as the intended final 3.x release, and we're working toward a 4.0 release in a few months. I'm closing this, but can re-open if you have some feedback.

IMO... your best bet may be to create an issue for the 4.0 release, which identifies your need to use the engine in an engine extension. If you are able to create a simplified example, using any extension type, which requires an engine instance we can figure out some way (in the new API) to give you that instance rather than requiring a reference. A reference will never work, IMO, since it won't give you the same instance under which the tests are running.
